certora/specs/HashConsensus.spec

```diff
@@ -0,0 +1,742 @@
+//  4. setFastLaneLengthSlots() works as expected, setting it to zero disables the fast lane subset
+//     verify with getFrameConfig() and check that fastLaneLengthSlots < frameConfig.epochsPerFrame * SLOTS_PER_EPOCH
+//  5. addMember() - cannot add an existing member
+//  6. addMember() - cannot add an empty address as a member
+//  7, addMember() - adding a member does not add/remove any other member
+//  8. addMember() - adding a member increases the total members by 1
+//  9. removeMember() - cannot remove a member that does not exists - reverts
+// 10. removeMember() - removing a member does not add/remove any other member
+// 11. removeMember() - removing a member decreases the total members by 1
+// 12. setQuorum() - acts as expectedly, verified by getQuorum()
+// 13. disableConsensus() - acts as expectedly, verified by getQuorum()
+// 14. setReportProcessor() - acts as expectedly, verified by getReportProcessor(), also cannot set an empty address or the previous address
+// 15. submitReport() - slot > max_uint64 => revert
+// 16. submitReport() - slot != refSlot of current frame => revert
+// 17. submitReport() - same member cannot submit two different reports for the same slot
+// 18. submitReport() - reportHash == 0 => revert
+// 19. submitReport() - consensusVersion != getConsensusVersion() => revert
+// 20. submitReport() - verify that the deadline (frame.reportProcessingDeadlineSlot) is explicitly ==  refSlot + FrameSize
+// 21. submitReport() - the same oracle should not report the same slot+hash twice (cannot double vote for same report)
+// 22. submitReport() - single call to submit report increases only one support for one variant
+//                      sum of supports of all variants < total members,
+//                      because if Oracle member changes its mind, its previous support is removed
+// 23. submitReport() - when new frame starts _reportVariantsLength should reset to 1
+// 24. updateInitialEpoch() - updates correctly the initialEpoch as returned by getFrameConfig()
+//                          - you cannot update the initialEpoch to be one that already it arrived
+// 25. initialEpochSanity rule - once the system is initialized its initialEpoch setting is always sane
+
+
+//  1. All the external view functions cannot revert under any circumstance,
+//     especially getCurrentFrame() as it is used by LegacyOracle and BaseOracle
+// Status: Fail
+// https://prover.certora.com/output/80942/ed36de90af05459d9eca30a2adbe4212/?anonymousKey=c9c730984b9637f6fc5d7700d5fc9f69dff5b4f6
+rule viewFunctionsDoNotRevert(method f)
+    filtered { f -> f.isView }
+{
+    env e; calldataarg args;
+    require e.block.timestamp > saneTimeConfig();
+    require e.msg.value == 0; // view functions revert is you send eth
+
+    f@withrevert(e,args);
+    assert !lastReverted;
+}
+
+// focus only on getCurrentFrame() and verify it does not revert
+// Status: Timeout! (without the simplification, shown in saneTimeConfig())
```

Comment:
You could have increased the time out settings even more, but the simplification is OK I guess.